python/pyspark/sql/tests/test_types.py

```diff
@@ -739,6 +739,11 @@ def test_timestamp_microsecond(self):
         tst = TimestampType()
         self.assertEqual(tst.toInternal(datetime.datetime.max) % 1000000, 999999)
 
+    # regression test for SPARK-23299
+    def test_row_without_column_name(self):
+        row = Row("Alice", 11)
```

Comment:
https://github.com/apache/spark/pull/20503#discussion_r225844655

Reply:
Indeed, this form of creating Row objects is probably rarely used and should at least be documented.  Maybe it wasn't even intended to be used in the way I did.  Please have a look at the docstring change I've added and let me know if this is appropriate.